python/ray/autoscaler/azure/node_provider.py

```diff
@@ -102,12 +102,19 @@ def _extract_metadata(self, vm):
             resource_group_name=resource_group,
             network_interface_name=metadata["nic_name"])
         ip_config = nic.ip_configurations[0]
-        public_ip_id = ip_config.public_ip_address.id
-        metadata["public_ip_name"] = public_ip_id.split("/")[-1]
-        public_ip = self.network_client.public_ip_addresses.get(
-            resource_group_name=resource_group,
-            public_ip_address_name=metadata["public_ip_name"])
-        metadata["external_ip"] = public_ip.ip_address
+
```

Comment:
are we sure this is needed? I thought the head node uses internal_ips normally?

Reply:
not sure if it's needed. just to make sure it works in all cases.